vraptor-core/src/main/java/br/com/caelum/vraptor/http/iogi/IogiParametersProvider.java

```diff
@@ -90,11 +90,12 @@ private Object instantiateOrAddError(Parameters parameters, List<Message> errors
 	private List<Target<Object>> createTargets(ControllerMethod method) {
 		Method javaMethod = method.getMethod();
 		List<Target<Object>> targets = new ArrayList<>();
+		TypeExtractor typeExtractor = new TypeExtractor(method.getController().getType());
```

Comment:
In the other hand, extracting to a class are giving us the single responsibility principle.
It's a small class, easily testable, with a well-defined purpose. That's why I'm ok with
this change. But sure, we can keep it as a private method. @francofabio it's your call.